### ISSUE TITLE
Improve Typescript dispatch reducers inference

### ISSUE THREAD

**Issue by semoal** [rematch/rematch#894]
```ts
import { createModel } from '@rematch/core'
import { delay } from './utils'
import type { RootModel } from '.'

export const dolphins = createModel<RootModel>()({
	state: 0,
	reducers: {
		increment: (state, payload = 1) => state + payload,
	}
})

```
**Describe the bug**

Actually the `payload = 1` is infered as any and when dispatching the reducer says that: `(property) increment: () => Action<void, void>`, doesn't expect a payload...

The same occurs if the payload it's optional

**Comment by tianzhich** [comment]
I've fixed it if the payload is optional.

**Comment by tianzhich** [comment]
Maybe it is [design limitation](https://github.com/microsoft/TypeScript/issues/41668).

**Comment by semoal** [comment]
Close this, since will be fixed with 2.1.0 release
